CONFIG_HIGHMEM与上游不一致，该单导致Sv48及以上的虚拟地址格式不可用

需要注意，这一单争议较大，上游一致没有进入HIGHMEM功能，这一单功能不完善，会导致Sv48及以上的虚拟地址格式不可用
https://github.com/RVCK-Project/rvck/commit/bfa3b88145be7fcdf0910787fa1f50f6e02505e6

**revert了几单**：
commit b6481228d265cd0cebb28d3c5606f0f0d715951e
riscv:mm: Use old __set_pte_at() because highmem is in use
https://github.com/RVCK-Project/rvck/commit/b6481228d265cd0cebb28d3c5606f0f0d715951e

commit bfa3b88145be7fcdf0910787fa1f50f6e02505e6
riscv: mm: Add high memory on riscv64 using sv39
https://github.com/RVCK-Project/rvck/commit/bfa3b88145be7fcdf0910787fa1f50f6e02505e6

commit eae5019659ca721e4817932f9a507996af0adc64
riscv: Stop emitting preventive sfence.vma for new userspace mappings with Svvptc
https://github.com/RVCK-Project/rvck/commit/eae5019659ca721e4817932f9a507996af0adc64

**cherry了新单**：
https://github.com/torvalds/linux/commit/01261e24cfab69c65043e1e61168348ae23a64c2

**遗留一些符号问题待查**
` make ARCH=riscv CROSS_COMPILE=riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu- -j256
  CALL    scripts/checksyscalls.sh
  CHK     kernel/kheaders_data.tar.xz
  UPD     include/generated/utsversion.h
  CC      init/version-timestamp.o
  LD      .tmp_vmlinux.kallsyms1
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: arch/riscv/kernel/entry.o: in function `_new_vmalloc_kernel_address':
(.irqentry.text+0x44): undefined reference to `new_vmalloc'
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: mm/percpu.o: in function `pcpu_populate_chunk':
percpu.c:(.text+0x285a): undefined reference to `new_vmalloc'
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: mm/vmalloc.o: in function `ioremap_page_range':
vmalloc.c:(.text+0x4b44): undefined reference to `new_vmalloc'
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: mm/vmalloc.o: in function `vm_map_ram':
vmalloc.c:(.text+0x5a22): undefined reference to `new_vmalloc'
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: mm/vmalloc.o: in function `vmap':
vmalloc.c:(.text+0x67bc): undefined reference to `new_vmalloc'
riscv64_gcc14.1.0_glibc2.39.0_binutils2.42/bin/riscv64-unknown-linux-gnu-ld: mm/vmalloc.o:vmalloc.c:(.text+0x6f60): more undefined references to `new_vmalloc' follow
make[2]: *** [scripts/Makefile.vmlinux:37: vmlinux] Error 1
make[1]: *** [rvck/Makefile:1172: vmlinux] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:234: __sub-make] Error 2`

--- a/mm/memblock.c
+++ b/mm/memblock.c
@@ -1733,7 +1733,6 @@ phys_addr_t __init_memblock memblock_end_of_DRAM(void)
 
 static phys_addr_t __init_memblock __find_max_addr(phys_addr_t limit)
 {
-	phys_addr_t phys_ram_base = memblock_start_of_DRAM();
 	phys_addr_t max_addr = PHYS_ADDR_MAX;
 	struct memblock_region *r;
 
@@ -1743,10 +1742,11 @@ static phys_addr_t __init_memblock __find_max_addr(phys_addr_t limit)
 	 * of those regions, max_addr will keep original value PHYS_ADDR_MAX
 	 */
 	for_each_mem_region(r) {
-		if ((r->base + r->size) >= (phys_ram_base + limit)) {
-			max_addr = phys_ram_base + limit;
+		if (limit <= r->size) {
+			max_addr = r->base + limit;
 			break;
 		}
+		limit -= r->size;
 	}
 
 	return max_addr;

